scripts/globals/spells/trust/kupipi.lua

```diff
@@ -22,19 +26,28 @@ function onSpellCast(caster, target, spell)
 end
 
 function onMobSpawn(mob)
-    mob:addListener("COMBAT_TICK", "KUPIPI_COMBAT_TICK", function(trust, master, target)
-        function round(x)
-            return x >= 0 and math.floor(x + 0.5) or math.ceil(x - 0.5)
-        end
-        
-        local party = master:getPartyWithTrusts()
-        local lvl = trust:getMainLvl()
-        local maxCure = utils.clamp(round(lvl/8), 1, 6)
-
-        for _, member in ipairs(party) do
-            if member:getHPP() <= 75 then
-                trust:castSpell(maxCure, member)
-            end
-        end
-    end)
+    local CURE_I = 1
+    local FLASH  = 112
+    local ERASE  = 143
+    local DISPEL = 260
+
+    mob:addGambit(PARTY, HPP_LTE, 25, MA, SELECT_HIGHEST, tpz.magic.spellFamily.CURE)
+
+    mob:addGambit(PARTY, STATUS, tpz.effect.SLEEP_I, MA, SELECT_SPECIFIC, CURE_I)
+    mob:addGambit(PARTY, STATUS, tpz.effect.SLEEP_II, MA, SELECT_SPECIFIC, CURE_I)
+
+    mob:addGambit(PARTY, HPP_LTE, 75, MA, SELECT_HIGHEST, tpz.magic.spellFamily.CURE)
+
+    mob:addGambit(PARTY, NOT_STATUS, tpz.effect.PROTECT, MA, SELECT_HIGHEST, tpz.magic.spellFamily.PROTECTRA)
+    mob:addGambit(PARTY, NOT_STATUS, tpz.effect.SHELL, MA, SELECT_HIGHEST, tpz.magic.spellFamily.SHELLRA)
+
+    mob:addGambit(SELF, STATUS_FLAG, tpz.effectFlag.ERASABLE, MA, SELECT_SPECIFIC, ERASE)
+    mob:addGambit(PARTY, STATUS_FLAG, tpz.effectFlag.ERASABLE, MA, SELECT_SPECIFIC, ERASE)
+
+    mob:addGambit(TARGET, STATUS_FLAG, tpz.effectFlag.DISPELABLE, MA, SELECT_SPECIFIC, DISPEL)
```

Comment:
Does Kupipi dispel?

Reply:
Nope! I think I got confused when trying to put in as much functionality as possible. She does -na though, so I'll swap those in